middleware/proxy/upstream.go

```diff
@@ -21,6 +22,14 @@ var (
 	supportedPolicies = make(map[string]func() Policy)
 )
 
+type upstreamEncoding uint
+
+const (
+	encodingNone upstreamEncoding = iota
+	encodingGRPC
```

Comment:
It's just easier to test without the certs, etc. I can pull it now though as I have tested the TLS path.